src/sources/kubernetes_logs/mod.rs

```diff
@@ -508,6 +543,50 @@ mod tests {
         crate::test_util::test_generate_config::<Config>();
     }
 
+    #[test]
+    fn prepare_exclude_paths() {
+        let cases = vec![
+            (
+                Config::default(),
+                vec![
+                    glob::Pattern::new("**/*.gz").unwrap(),
+                    glob::Pattern::new("**/*.tmp").unwrap(),
+                ],
+            ),
+            (
+                Config {
+                    exclude_paths_glob_patterns: vec![std::path::PathBuf::from("**/*.tmp")],
+                    ..Default::default()
+                },
+                vec![glob::Pattern::new("**/*.tmp").unwrap()],
+            ),
+            (
+                Config {
+                    exclude_paths_glob_patterns: vec![
+                        std::path::PathBuf::from("**/kube-system_*/**"),
+                        std::path::PathBuf::from("**/*.gz"),
+                        std::path::PathBuf::from("**/*.tmp"),
+                    ],
+                    ..Default::default()
+                },
+                vec![
+                    glob::Pattern::new("**/kube-system_*/**").unwrap(),
+                    glob::Pattern::new("**/*.gz").unwrap(),
+                    glob::Pattern::new("**/*.tmp").unwrap(),
+                ],
+            ),
+        ];
+
+        for (input, mut expected) in cases {
+            let mut output = super::prepare_exclude_paths(&input).unwrap();
+            assert_eq!(
+                expected.sort(),
+                output.sort(),
```

Comment:
Is this right? I don't think `sort()` returns anything: https://doc.rust-lang.org/std/vec/struct.Vec.html#method.sort

Reply:
It seemed to fail otherwise, if the paths were in different orders 🤔 I'll re-verify

Reply:
Right, I mean I think sorting is correct here. I'm just surprised that you could do it directly here since the functions don't seem to return anything so I think it'd just be comparing `()` with `()`.

Reply:
Isn't that what's happening in the example you shared? Since it's mutable doesn't it update in place?

Reply:
```
error: `assert_eq` of unit values detected. This will always succeed
```
🤷